conjure-java-core/src/main/java/com/palantir/conjure/java/services/dialogue/DialogueEndpointsGenerator.java

```diff
@@ -37,20 +37,27 @@
 import com.squareup.javapoet.TypeName;
 import com.squareup.javapoet.TypeSpec;
 import java.util.Map;
+import java.util.Optional;
 import javax.lang.model.element.Modifier;
 import org.glassfish.jersey.uri.internal.UriTemplateParser;
 
 final class DialogueEndpointsGenerator {
     private final Options options;
-    private final String apiVersion;
+    private final Optional<String> apiVersion;
 
-    DialogueEndpointsGenerator(Options options, String apiVersion) {
+    DialogueEndpointsGenerator(Options options, Optional<String> apiVersion) {
         this.options = options;
         this.apiVersion = apiVersion;
     }
 
     public JavaFile endpointsClass(ServiceDefinition def) {
         ClassName serviceClassName = Names.endpointsClassName(def, options);
+        CodeBlock versionBlock = apiVersion
+                .map(version -> CodeBlock.of("return $S;", apiVersion))
+                .orElseGet(() -> CodeBlock.of(
+                        "return Optional.ofNullable($T.class.getPackage().getImplementationVersion())"
+                                + ".orElse(\"0.0.0\");",
+                        serviceClassName));
```

Comment:
Can we store this to a static final field once per set of endpoints and avoid runtime cost?